src/System.Private.CoreLib/src/System/Threading/Mutex.Windows.cs

```diff
@@ -0,0 +1,143 @@
+                        if (errorCode == Interop.mincore.Errors.ERROR_SUCCESS)
+                        {
+                            errorCode = Interop.mincore.Errors.ERROR_ALREADY_EXISTS;
+                        }
+                        break;
+                    }
+                }
+                else
+                {
+                    break;
+                }
+            }
+            return errorCode;
+        }
+
+        private static OpenExistingResult OpenExistingWorker(string name, out Mutex result)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name), SR.ArgumentNull_WithParamName);
+            }
+
+            if (name.Length == 0)
+            {
+                throw new ArgumentException(SR.Argument_EmptyName, nameof(name));
+            }
+            if (((int)Interop.Constants.MaxPath) < (uint)name.Length)
+            {
+                throw new ArgumentException(SR.Format(SR.Argument_WaitHandleNameTooLong, Interop.Constants.MaxPath), nameof(name));
+            }
+            Contract.EndContractBlock();
+
+            result = null;
+
+            // To allow users to view & edit the ACL's, call OpenMutex
+            // with parameters to allow us to view & edit the ACL.  This will
+            // fail if we don't have permission to view or edit the ACL's.  
+            // If that happens, ask for less permissions.
+            SafeWaitHandle myHandle = new SafeWaitHandle(Interop.mincore.OpenMutex((uint)(Interop.Constants.MutexModifyState | Interop.Constants.Synchronize), false, name), true);
+
+            int errorCode = 0;
+            if (myHandle.IsInvalid)
+            {
+                errorCode = (int)Interop.mincore.GetLastError();
+
+                if (Interop.mincore.Errors.ERROR_FILE_NOT_FOUND == errorCode || Interop.mincore.Errors.ERROR_INVALID_NAME == errorCode)
+                    return OpenExistingResult.NameNotFound;
+                if (Interop.mincore.Errors.ERROR_PATH_NOT_FOUND == errorCode)
+                    return OpenExistingResult.PathNotFound;
+                if (null != name && 0 != name.Length && Interop.mincore.Errors.ERROR_INVALID_HANDLE == errorCode)
+                    return OpenExistingResult.NameInvalid;
+
+                // this is for passed through Win32Native Errors
+                throw ExceptionFromCreationError(errorCode, name);
+            }
+
+            result = new Mutex(myHandle);
+            return OpenExistingResult.Success;
+        }
+
+        // Note: To call ReleaseMutexCore, you must have an ACL granting you
+        // MUTEX_MODIFY_STATE rights (0x0001).  The other interesting value
+        // in a Mutex's ACL is MUTEX_ALL_ACCESS (0x1F0001).
+        private static void ReleaseMutexCore(IntPtr handle)
+        {
+            if (!Interop.mincore.ReleaseMutex(handle))
+            {
+                ThrowSignalOrUnsignalException();
+            }
+        }
+    }
+}
```

Comment:
I believe so, I did a diff a while back and there were some small differences here and there, but nothing significant. I'll take another look and make sure.